client/src/components/advancedSearch/TaskFilter.tsx

```diff
@@ -1,41 +1,237 @@
+        })
+      } else {
+        setExpandedItems(prev => new Set([...prev, task.uuid]))
+      }
+    }
+  }
+
+  const buildFlattenedList = (tasks, level = 0) => {
+    return tasks.flatMap(task => {
+      const isTaskSelected = selectedItems?.some(
+        item => item.uuid === task.uuid
+      )
+      const hasDescendantTasksSelected = selectedItems?.some(item =>
+        task.descendantTasks?.some(child => child.uuid === item.uuid)
+      )
+      const isCollapsed = !expandedItems.has(task.uuid)
+      const isSelected = isTaskSelected
+        ? true
+        : hasDescendantTasksSelected && isCollapsed
+          ? null
+          : false
+      const taskWithLevel = { ...task, level, isSelected }
+      const children = expandedItems.has(task.uuid)
+        ? childrenMap.get(task.uuid) || []
+        : []
+      return [taskWithLevel, ...buildFlattenedList(children, level + 1)]
+    })
+  }
+
+  const topLevelItems = items.filter(task => !task.parentTask)
+  const flattenedItems = buildFlattenedList(topLevelItems)
+
+  const enhancedRenderRow = task => {
+    const hasChildren = task.childrenTasks?.length > 0
+    const isExpanded = expandedItems.has(task.uuid)
+    const isSelected = selectedItems?.some(item => item.uuid === task.uuid)
+
+    const handleToggleSelection = e => {
+      e.stopPropagation()
+      if (isSelected) {
+        handleRemoveItem(task)
+      } else {
+        handleAddItem(task)
+      }
+    }
+
+    const displayLabel = task.longName
+      ? `${task.shortName}: ${task.longName}`
+      : task.shortName
+    const padding = Math.min(task.level, 3) * 20 + (hasChildren ? 0 : 26)
+    const indentedLabel = (
+      <div
+        style={{
+          display: "flex",
+          flexDirection: "row",
+          alignItems: "center",
+          paddingLeft: padding,
+          gap: 10,
+          cursor: "auto"
+        }}
+      >
+        {(hasChildren && (
+          <>
+            <span
+              onClick={e => {
+                e.stopPropagation()
+                handleExpand(task)
+              }}
+              style={{ cursor: "pointer" }}
+            >
+              <Icon
+                icon={
+                  isExpanded ? IconNames.CHEVRON_DOWN : IconNames.CHEVRON_RIGHT
+                }
+                size={IconSize.STANDARD}
+              />
+            </span>
+            <Icon
+              icon={isExpanded ? IconNames.FOLDER_OPEN : IconNames.FOLDER_CLOSE}
+              size={IconSize.STANDARD}
+            />
+          </>
+        )) || <Icon icon={IconNames.DOCUMENT} size={IconSize.STANDARD} />}
+        <img
+          src={TASKS_ICON}
+          alt=""
+          style={{ marginLeft: 5, marginRight: 5, height: "1em" }}
+        />
+        <span
+          onClick={handleToggleSelection}
+          style={{
+            cursor: "pointer",
+            flexGrow: "1"
+          }}
+        >
+          {displayLabel}
+        </span>
```

Comment:
Put the `<img>` inside the `<span>` below, so it is part of the "link" to the task that toggles the selection (which is more consistent with the link to a task in a LinkTo, as there the icon is also part of it).